### DISCUSSION TITLE
Support for Prettier PHP formatter

### DISCUSSION THREAD

**Discussion by adictonator** [tailwindlabs/tailwindcss#13031] [Ideas]
Hello community,

I would like to suggest that we add support for the Prettier PHP formatter along with the `prettier-plugin-tailwindcss` package. This would enable users to format their PHP-Tailwind CSS projects and ensure properly formatted and aligned Tailwind classes.

I hope the community will consider this suggestion and work towards implementing support for the Prettier PHP formatter with `prettier-plugin-tailwindcss`. Thank you for your time and consideration.

Best regards,
Aditya

**Comment by jivemedia** [comment]
This would be great - much needed feature!

**Comment by runtime-studio-au** [comment]
Hi @adictonator,

I commented on this two years ago, wanting the same thing, and ended up building it under my other business.

The plugin itself can't support PHP (Prettier has no PHP parser, and mixed PHP/HTML can't be parsed into a single tree), so this is a standalone CLI that lexes the file instead and sorts only the `class="..."` values — using the official engine via `prettier-plugin-tailwindcss/sorter`, so the order matches Prettier exactly. Nothing else in the file is touched.

[https://github.com/runtime-studio-au/tailwind-sort-php](https://github.com/runtime-studio-au/tailwind-sort-php)

Disclosure: I'm the author. MIT — edge cases welcome as issues.
